### PR TITLE
feat: Add expected valid app hash

### LINKED ISSUE #11: Add expected valid app hash to Fraud Proof
We'd like to add the expected valid app hash in Fraud Proof struct so that during verification, a light client knows what is the valid app hash that it should expect.

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Description

We'd like to add the expected valid app hash in Fraud Proof struct so that during verification, a light client knows what is the valid app hash that it should expect.

Closes: #11 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: baseapp/abci.go [hunk 1]
```diff
@@ -304,7 +304,7 @@ func (app *BaseApp) VerifyFraudProof(req abci.RequestVerifyFraudProof) (res abci
 		}
 
 		appHash = appFromFraudProof.GetAppHash(abci.RequestGetAppHash{}).AppHash
-		success = bytes.Equal(appHash, req.ExpectedAppHash)
+		success = bytes.Equal(appHash, req.ExpectedValidAppHash)
```
Comment (tzdybal) [inline baseapp/abci.go:307]: `VerifyFraudProof` should return `true` if and only if fraud proof is completely valid and proves that fraudulent state transition happened.
`appHash` is generated locally, so we must trust it.
`req.ExpectedValidAppHash` was received from P2P network, so we're not sure it's correct. By comparing here we ensure that received fraud proof was honest.

Comment (tzdybal) [reply baseapp/abci.go:307]: Ensuring that block actually contained different (invalid) ISR/appHash, is a separate issue - do we have one in GitHub?

**Review by tzdybal — APPROVED** [review]
LGTM

File: baseapp/abci.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply baseapp/abci.go:307]: Just made one here: https://github.com/celestiaorg/cosmos-sdk-rollmint/issues/12

**Pull request merged**